azkaban-web-server/src/main/java/azkaban/imagemgmt/daos/ImageVersionDaoImpl.java

```diff
@@ -0,0 +1,132 @@
+      String currentTime = DateTime.now().toLocalDateTime().toString();
+      transOperator.update(INSERT_IMAGE_VERSION_QUERY, imageVersion.getPath(), imageVersion.getDescription(),
+          imageVersion.getVersion(), imageType.getId(), imageVersion.getState().getStateValue(),
+          imageVersion.getReleaseTag(), currentTime, imageVersion.getCreatedBy());
+      transOperator.getConnection().commit();
+      return transOperator.getLastInsertId();
+    };
+
+    int imageVersionId = 0;
+    try {
+      /* what will happen if there is a partial failure in
+         any of the below statements?
+         Ideally all should happen in a transaction */
+      imageVersionId = databaseOperator.transaction(insertAndGetSpaceId).intValue();
+    } catch (SQLException e) {
+      log.error("Unable to create the image version metadata", e);
+    }
+    return imageVersionId;
+  }
+
+  @Override
+  public List<ImageVersion> getImageVersion(RequestContext requestContext) throws ImageMgmtException {
+    List<ImageVersion> imageVersions = new ArrayList<>();
+    try {
+      StringBuilder queryBuilder = new StringBuilder(SELECT_IMAGE_VERSION_BASE_QUERY);
+      if(requestContext.getParams().containsKey(ImageMgmtConstants.IMAGE_TYPE)) {
+        queryBuilder.append(" AND ");
+        queryBuilder.append(" it.type = '"+requestContext.getParams().get(ImageMgmtConstants.IMAGE_TYPE)+"'");
+      }
+
+      if(requestContext.getParams().containsKey(ImageMgmtConstants.IMAGE_VERSION)) {
+        queryBuilder.append(" AND ");
+        queryBuilder.append(" iv.version = '"+requestContext.getParams().get(ImageMgmtConstants.IMAGE_VERSION)+"'");
+      }
+      log.info("Query : "+queryBuilder.toString());
+      imageVersions = databaseOperator.query(queryBuilder.toString(),
+          new FetchImageVersionHandler());
+    } catch (SQLException ex) {
+      log.error("Exception while fetching image version ", ex);
+      throw  new ImageMgmtException("Exception while fetching image version");
+    }
+    return imageVersions;
+  }
+
+  public static class FetchImageVersionHandler implements ResultSetHandler<List<ImageVersion>> {
+    @Override
+    public List<ImageVersion> handle(final ResultSet rs) throws SQLException {
+      if (!rs.next()) {
+        return Collections.emptyList();
+      }
+      final List<ImageVersion> imageVersions = new ArrayList<>();
+      do {
+        final int id = rs.getInt("id");
+        final String path = rs.getString("path");
+        final String description = rs.getString("description");
+        final String version = rs.getString("version");
+        final String type = rs.getString("type");
+        final String state = rs.getString("state");
+        final String releaseTag = rs.getString("release_tag");
+        final String createdOn = rs.getString("created_on");
+        final String createdBy = rs.getString("created_by");
+        final String modifiedOn = rs.getString("modified_on");
```

Comment:
Why is type of date parameter String?

Reply:
Created and updated timestamp is directly inserted from database during insert/update. The corresponding datatype for them in database is timestamp. So, from the code these fields are never populated. These fields are only retrieved as it is in the code and formatted in string. Hence, in the code datetime datatype is not used.